### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by sthayashi**
This is an initial commit. Tests will be added soon.

### CODE REVIEW

File: test/test_array_shape.jl [hunk 1]
```diff
@@ -64,16 +66,82 @@ using ElasticArrays, ArraysOfArrays
...
+        @test @inferred(getindex(A, ac1, ac1, ac1)) == A
+        @test @inferred(getindex(A, ac2, ac2, ac2))[1] == A[1]
+        @test @inferred(getindex(A, ac4, ac4, ac4))[1] == 63
+
+        @test @inferred(view(A, ac1, ac1, ac1)) == A
+        @test @inferred(view(A, ac2, ac2, ac2))[1] == A[1]
+        @test @inferred(view(A, ac4, ac4, ac4))[1] == 63
+
+        first_layer = @inferred(getindex(A, ac1, ac1, ac2))
+        setindex!(A, first_layer, ac1, ac1, ac3)
```
Comment (sthayashi) [inline test/test_array_shape.jl:91]: Note that to use setindex! with ValueAccessors:

```
# Replace the scalar type element of A[1] = 1 with A[2] = 2
v = getindex(A, ac2, ac2, ac2).+1

# With and Without ValueAccessor
setindex!(A, v, ac2, ac2, ac2)
setindex!(A, 2, 1, 1, 1)

# Doesn't work
setindex!(A, 2, ac2, ac2, ac2)
```

Comment (sthayashi) [reply test/test_array_shape.jl:91]: These getindex calls should also be showing coverage on line 86 in array_shape.jl, but aren't.

Comment (oschulz) [reply test/test_array_shape.jl:91]: Hm, no idea why.

**Comment by oschulz** [comment]
I've merged this, so we can see better where we are, and so I can use the new tests when I make changes. We can also make another PR to improve test coverage further.

**Comment by sthayashi** [comment]
Sounds good to me. I'm just looking at why the _rand! and _logpdf function in const_value_dist.jl aren't working. I'm wondering if it's due to the way it broadcasts.

**Pull request merged**